gnosis/safe/tests/test_safe.py

```diff
@@ -815,3 +815,21 @@ def test_send_previously_approved_tx(self):
 
         balance = self.w3.eth.get_balance(to)
         self.assertEqual(value, balance)
+
+    def test_get_contracts_cache_leak(self):
```

Comment:
np: maybe the `leak` in the name indicates this wasn't working well previously, but I'm not sure it describes the unit test purpose.

Reply:
The purpouse of the test is check that the cache storage make sense regarding the function calls. Then maybe could be better `test_cache_size_get_contracts`. What do you think?

Reply:
Yeah I agree that having `leak` in the test could indicate that you are testing that if it leaks or not.

The test should be more like "given a cache size of X, storing more than X would still keep the size at X size"

Then you can test the `max_size`. For testing purposes you can try having a really small `max_size` and see how it behaves.

Reply:
@fmrsabino I was trying to add test for maxsize but I couldn't because it is defined in the decorator.   
Anyway I don't think that test maxsize is necessary because is part of the unit test on the library.

Reply:
> @fmrsabino I was trying to add test for maxsize but I couldn't because it is defined in the decorator. Anyway I don't think that test maxsize is necessary because is part of the unit test on the library.

What I meant was not to test `max_size` directly but that this function uses some "max size". But probably this means that we need to read some internals of `cached` and that would kind of defeat the purpose of the test (it would be too tied to the implementation).

An example can be seen [here](https://docs.python.org/3/library/functools.html)
```python
@lru_cache(maxsize=32)
def get_pep(num):
    'Retrieve text of a Python Enhancement Proposal'
    resource = 'https://www.python.org/dev/peps/pep-%04d/' % num
    try:
        with urllib.request.urlopen(resource) as s:
            return s.read()
    except urllib.error.HTTPError:
        return 'Not Found'

>>> for n in 8, 290, 308, 320, 8, 218, 320, 279, 289, 320, 9991:
...     pep = get_pep(n)
...     print(n, len(pep))

>>> get_pep.cache_info()
CacheInfo(hits=3, misses=8, maxsize=32, currsize=8)
```

I think maybe a test could be that **it has as `maxsize`**. This way if this decorator is removed the test would fail (by using `get_contract.cache_info()`.

What do you think?

Reply:
Yes it's true we could use `get_contract.cache.maxsize` but it's the same It's going to break if we remove the decorator.   
Maybe I could add a test using `tracemalloc` that takes a snapshot at the beginning of call this function and other at the end and check the memory don't pass of some value. 
This would be independent of the decorator.  :thinking:

Reply:
> Yes it's true we could use `get_contract.cache.maxsize` but it's the same It's going to break if we remove the decorator. Maybe I could add a test using `tracemalloc` that takes a snapshot at the beginning of call this function and other at the end and check the memory don't pass of some value. This would be independent of the decorator. 🤔

But it will break already because your test accesses the `cache` too. I think this is ok! If we remove the cache then the test should fail. The specification for this function is that it is cached and we are testing that.

My suggestion was more to test that you have a `max_size` set 👍 (ie.: it's not `None`)